src/original-content/VirtualFileSystem.cpp

```diff
@@ -89,11 +89,37 @@ bs::Vector<bs::String> VirtualFileSystem::listAllFiles()
   for (size_t i = 0; i < allStl.size(); i++)
   {
     all[i] = allStl[i].c_str();
+
+    // Internal file index will return the files in the casing they were stored in.
+    // To be consistent, convert them all to uppercase here.
+    bs::StringUtil::toUpperCase(all[i]);
   }
 
   return all;
 }
 
+bs::Vector<bs::String> REGoth::VirtualFileSystem::listByExtension(const bs::String& ext)
+{
+  bs::Vector<bs::String> all = listAllFiles();
+
+  enum
+  {
+    RespectCase = false,
+    LowerCase  = true,
+  };
+
+  bs::Vector<bs::String> result;
+  for(const auto& f : all)
+  {
+    if (bs::StringUtil::endsWith(f, ext, RespectCase))
```

Comment:
Why is the case respected in this case? I think it's something which can cause a lot of confusion and is very unintuitive for people who don't carefully read the documentation of this method.

Reply:
All file names searched there are already in uppercase, as all methods from the virtual file system work with those. I chose to not use `ConvertToLower` here for efficiency, because it would then have to convert the filenames to lowercase as well. However, converting the extension to uppercase before the loop would work I guess.

Reply:
Ah, I probably looked to quickly and didn't notice you are using the `endsWith` method (thought you were just plainly comparing strings and `f` is the extension). Converting the extension should work best, yes (I wouldn't worry so much about efficieny yet)